src/tiling/tile_state.rs

```diff
@@ -62,10 +63,31 @@ pub struct TileStateMut<'a, T: Pixel> {
   pub restoration: TileRestorationStateMut<'a>,
   pub half_res_pmvs: &'a mut Vec<BlockPmv>,
   pub mvs: Vec<TileMotionVectorsMut<'a>>,
+  pub coded_block_info: Vec<Vec<CodedBlockInfo>>, // indexed in MI units
```

Comment:
It could be a `Box<[CodedBlockInfo]>` maybe?

Reply:
Changed to a new structure with a single vector and `Index`/`IndexMut` implementations instead.